### ISSUE TITLE
`public_member_api_docs ` should also highlight library statements.

### ISSUE THREAD

**Issue by sigurdm** [dart-lang/linter#4962]
See https://github.com/dart-lang/pana/issues/1355 for a bit of discussion

<img width="819" alt="image" src="https://github.com/dart-lang/linter/assets/8613953/f013c7b7-2b87-47a8-a647-017b8cc6b8d4">

**Comment by jonasfj** [comment]
This should only apply to `library` statements in public libraries.
Meaning, it doesn't have to apply to files in:
 * `bin/`,
 * `lib/src/`
 * `test/`
 * `example/`
 * `hook/`

Basically, only files in `lib/` excluding `lib/src/`.

-----------------------

It's also entirely reasonable that we should complain about missing library statement.
Because all libraries will have a documentation page, even if they don't have a library statement.

**Comment by srawlins** [comment]
As this would be a massively breaking, impossible-to-land change, I think this will take the shape of a new lint rule 😄

**Comment by bwilkerson** [comment]
I agree, a new lint is probably the best path forward, maybe something like `public_library_api_docs`. I'm not sure I'd restrict it to public libraries that have a `library` directive, though. Seems like all public libraries should have docs?
